sentry_sdk/integrations/openai.py

```diff
@@ -0,0 +1,228 @@
+        messages = kwargs["messages"]
+        model = kwargs.get("model")
+        streaming = kwargs.get("stream")
+
+        span = hub.start_span(op="openai", description="Chat Completion")
+        span.__enter__()
+        res = f(*args, **kwargs)
+        with capture_internal_exceptions():
+            span.set_data("messages", messages)
+            span.set_tag("model", model)
+            span.set_tag("streaming", streaming)
+
+            if hasattr(res, "choices"):
+                span.set_data("response", res.choices[0].message)
+                _calculate_chat_completion_usage(messages, res, span)
+                span.__exit__(None, None, None)
+            elif hasattr(res, "_iterator"):
+                data_buf: list[list[str]] = []  # one for each choice
+
+                old_iterator = res._iterator  # type: Iterator[ChatCompletionChunk]
+
+                def new_iterator():
+                    # type: () -> Iterator[ChatCompletionChunk]
+                    with capture_internal_exceptions():
+                        for x in old_iterator:
+                            if hasattr(x, "choices"):
+                                choice_index = 0
+                                for choice in x.choices:
+                                    if hasattr(choice, "delta") and hasattr(
+                                        choice.delta, "content"
+                                    ):
+                                        content = choice.delta.content
+                                        if len(data_buf) <= choice_index:
+                                            data_buf.append([])
+                                        data_buf[choice_index].append(content or "")
+                                    choice_index += 1
+                            yield x
+                        if len(data_buf) > 0:
+                            all_responses = list(
+                                map(lambda chunk: "".join(chunk), data_buf)
+                            )
+                            span.set_data("responses", all_responses)
+                            _calculate_chat_completion_usage(
+                                messages, res, span, all_responses
+                            )
+                    span.__exit__(None, None, None)
+
+                res._iterator = new_iterator()
+            else:
+                span.set_tag("unknown_response", True)
+                span.__exit__(None, None, None)
+            return res
+
+    return new_chat_completion
+
+
+def _wrap_enbeddings_create(f):
+    # type: (Callable[..., Any]) -> Callable[..., Any]
+
+    @wraps(f)
+    def new_embeddings_create(*args, **kwargs):
+        # type: (*Any, **Any) -> Any
+        hub = Hub.current
+        integration = hub.get_integration(OpenAIIntegration)
+        if integration is None:
+            return f(*args, **kwargs)
+
+        with hub.start_span(op="openai", description="Embeddings Creation") as span:
```

Comment:
Also same as mentioned above. Please use `sentry_sdk.start_span(...)` instead of calling it from the Hub. We are in the process of removing the hub and this way it will be forward compatible.